### ISSUE TITLE
How long should our secret be when generating the TOTP secret?

### ISSUE THREAD

**Issue by forevermatt** [silinternational/totp-api#15]
@fillup / @wcjr / @Baggerone

**tl;dr** - What balance do we want to strike between longer TOTP Key length (= more protection against... what?) vs. shorter code for a user to (possibly) type in?
- default = 52 chars
- Daplie demo = 32 chars
- Authy in-app example = 16 chars
- Slack = 16 chars
---

I had been using the default (`{length: 32}`), giving a TOTP Key (base32) string that is 52 characters long. That's really painful to try to type in (which is an option for users that can't scan a QR code).

The [daplie demo](https://daplie.github.io/browser-authenticator/) generates a TOTP key 32 characters long, which seems to be a TOTP secret length of 20.

When manually typing the code into Authy, the example they give is 16 characters long, so a TOTP secret length of 10.

Slack also produces a code 16 characters long (so `{length: 10}`).

My inclination is to follow Slack's example, and generate codes that are 16 - 20 characters long.

**Comment by wcjr** [comment]
fwiw, Google's is 32...

I'd be fine with 16.
